tests/test_dynamicbatching.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Dict, Any
+
+import unittest
+from parlai.core.opt import Opt
+import parlai.utils.testing as testing_utils
+from parlai.tasks.integration_tests.agents import NUM_TEST, EXAMPLE_SIZE
+
+_TASK = 'integration_tests:variable_length'
+_DEFAULT_OPTIONS = {
+    'batchsize': 8,
+    'dynamic_batching': 'full',
+    'optimizer': 'adamax',
+    'learningrate': 7e-3,
+    'num_epochs': 1,
+    'n_layers': 1,
+    'n_heads': 1,
+    'ffn_size': 32,
+    'embedding_size': 32,
+    'truncate': 8,
+    'model': 'transformer/generator',
+    'task': _TASK,
+}
+
+
+# TODO tests to write:
```

Comment:
nit: looks like you can get rid of this TODO?

Reply:
i think it's not complete, i'd rather keep it.